flight/Modules/OnScreenDisplay/onscreendisplay.c

```diff
@@ -0,0 +1,1448 @@
+/**
+ ******************************************************************************
+ * @addtogroup auLabsModules Tau Labs Modules
+ * @{
+ * @addtogroup OnScreenDisplay OSD Module
+ * @brief Process OSD information
+ * @{
+ *
+ * @file       onscreendisplay.c
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2013-2014
+ * @author     The OpenPilot Team, http://www.openpilot.org Copyright (C) 2010-2014.
+ * @author     dRonin, http://dronin.org Copyright (C) 2015
+ * @brief      OSD gen module, handles OSD draw. Parts from CL-OSD and SUPEROSD projects
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+// ****************
+
+// #define DEBUG_TIMING
+// #define DEBUG_ALARMS
+// #define DEBUG_TELEMETRY
+// #define DEBUG_BLACK_WHITE
+// #define DEBUG_STUFF
+#define SIMULATE_DATA
+#define TEMP_GPS_STATUS_WORKAROUND
+
+// static assert
+#define ASSERT_CONCAT_(a, b) a##b
+#define ASSERT_CONCAT(a, b) ASSERT_CONCAT_(a, b)
```

Comment:
What's the point of the double (and redundant?) MACRO for concatenation? Could that be accomplished in just one MACRO?